### ISSUE TITLE
Weird Texts on Screen

### ISSUE THREAD

**Issue by CanerKaraca23** [cleolibrary/CLEO5#252]
There is some weird texts on screen (i think it shows coordinats and it says ini file loaded on game opening screen), this happened after new RC release. I tried to delete cleo_tests folder (why this even exists on release package, is it related to debugging/development things?) and disabled logging on debug plugin .ini on plugins folder and still exists, it's annoying.

![sa-mp-010](https://github.com/user-attachments/assets/b7b82523-d574-4366-a708-5ea79d6d49a0)

**Comment by MiranDMC** [comment]
Looks like something from SAMP itself or other mods (like display area names around radar). Clearly intentional texts, not some random data from memory.

Test sources act as examples for each opcode as way to debug on user's side. These can not be executed without compiling them first.

**Comment by CanerKaraca23** [comment]
> Looks like something from SAMP itself or other mods (like display area names around radar). Clearly intentional texts, not some random data from memory.
> 
> Test sources act as examples for each opcode as way to debug on user's side. These can not be executed without compiling them first.

Yeah, display area, FPS and stream usage things are my mods. I mention this, this from CLEO RC1, Beta 1 works fine:

![sa-mp-010](https://github.com/user-attachments/assets/e3934fc4-188d-48b4-ae88-a8ada7b202ed)

@MiranDMC

**Comment by x87** [comment]
show cleo.log

**Comment by CanerKaraca23** [comment]
> show cleo.log

[.cleo.log](https://github.com/user-attachments/files/18211487/default.cleo.log)

@x87

**Comment by x87** [comment]
compass.cs? can you share this and other cleo scripts? one of them must have this code for on-screen coords

**Comment by CanerKaraca23** [comment]
> compass.cs? can you share this and other cleo scripts? one of them must have this code for on-screen coords

Sure, here all my cleo mods (you need CLEO+).

You should test with RC1 and Beta 1, Beta 1 works fine for me. @x87 

[cleo.zip](https://github.com/user-attachments/files/18211645/cleo.zip)

**Comment by MiranDMC** [comment]
Would be easier if you find the exact script causing it.

**Comment by MiranDMC** [comment]
> > Looks like something from SAMP itself or other mods (like display area names around radar). Clearly intentional texts, not some random data from memory.
> > Test sources act as examples for each opcode as way to debug on user's side. These can not be executed without compiling them first.
> 
> Yeah, display area, FPS and stream usage things are my mods. I mention this, this from CLEO RC1, Beta 1 works fine:
> 
> ![sa-mp-010](https://private-user-images.githubusercontent.com/37447503/397758783-e3934fc4-188d-48b4-ae88-a8ada7b202ed.jpg?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzQ3MDU4NTAsIm5iZiI6MTczNDcwNTU1MCwicGF0aCI6Ii8zNzQ0NzUwMy8zOTc3NTg3ODMtZTM5MzRmYzQtMTg4ZC00OGI0LWFlODgtYThhZGE3YjIwMmVkLmpwZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEyMjAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMjIwVDE0MzkxMFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTY2NjAyYjU1MjQ5MWY5NWFjMGUzYjk3ZjJiODRhYzVhMjFhN2JlOTY1NmFlYjBlMGIwNDdiNzM4NzRiYjNkMjAmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.6tyE5aIXDbYLTJKY9nAISDI0JA59ND79qqy5R8b4zQc)
> 
> @MiranDMC

Looks like some debug text. I would say from ambient mod. Coords, area name and some ambient params?

**Comment by CanerKaraca23** [comment]
> > > Looks like something from SAMP itself or other mods (like display area names around radar). Clearly intentional texts, not some random data from memory.
> > > Test sources act as examples for each opcode as way to debug on user's side. These can not be executed without compiling them first.
> > 
> > 
> > Yeah, display area, FPS and stream usage things are my mods. I mention this, this from CLEO RC1, Beta 1 works fine:
> > ![sa-mp-010](https://private-user-images.githubusercontent.com/37447503/397758783-e3934fc4-188d-48b4-ae88-a8ada7b202ed.jpg?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzQ3MDU4NTAsIm5iZiI6MTczNDcwNTU1MCwicGF0aCI6Ii8zNzQ0NzUwMy8zOTc3NTg3ODMtZTM5MzRmYzQtMTg4ZC00OGI0LWFlODgtYThhZGE3YjIwMmVkLmpwZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEyMjAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMjIwVDE0MzkxMFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTY2NjAyYjU1MjQ5MWY5NWFjMGUzYjk3ZjJiODRhYzVhMjFhN2JlOTY1NmFlYjBlMGIwNDdiNzM4NzRiYjNkMjAmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.6tyE5aIXDbYLTJKY9nAISDI0JA59ND79qqy5R8b4zQc)
> > @MiranDMC
> 
> Looks like some debug text. I would say from ambient sounds mod. Coords, area name and some ambient params?

Yeah, i just tested the mods one by one and it causes by SAAS (San Andreas Ambient Sounds) mod. I didn't uploaded this mod to .zip file because it's heavy (300mb). @MiranDMC @x87

**Comment by MiranDMC** [comment]
Your cleo log says it is not RC1 version. It does not list loaded ASI plugins, so it has to be older versions.

**Comment by CanerKaraca23** [comment]
> Your cleo log says it is not RC1 version. It does not list loaded ASI plugins, so it has to be older version.
> 
> If it is SAAS caused try rename it to *.cs4 I suspect this might be related to fixes in ini files handling opcodes.

I firstly tried RC1, when i see the issue, installed latest CI build so test if its fixed on latest commits but still same, maybe thats why not saying the version on .log file.

Renaming it to .cs4 fixed the issue, but yeah, it's definitely something wrong between RC1 and SAAS. @MiranDMC 

Mod Link: https://www.mixmods.com.br/2018/07/saas-san-andreas-ambient-sounds/

**Comment by x87** [comment]
Can you upload cs only?

**Comment by CanerKaraca23** [comment]
> Can you upload cs only?

Sure, here.

[SAAS.zip](https://github.com/user-attachments/files/18212200/SAAS.zip)

**Comment by MiranDMC** [comment]
> > Your cleo log says it is not RC1 version. It does not list loaded ASI plugins, so it has to be older version.
> > If it is SAAS caused try rename it to *.cs4 I suspect this might be related to fixes in ini files handling opcodes.
> 
> I firstly tried RC1, when i see the issue, installed latest CI build so test if its fixed on latest commits but still same, maybe thats why not saying the version on .log file.
> 
> Renaming it to .cs4 fixed the issue, but yeah, it's definitely something wrong between RC1 and SAAS. @MiranDMC
> 
> Mod Link: https://www.mixmods.com.br/2018/07/saas-san-andreas-ambient-sounds/

I guess it parses the input file in some way expecting old undefined behaviors from opcodes, which were fixed in CLEO5 but are still supported in legacy mode. In new mode data is probably not processed and Junior displays debug message with coords not covered by ambient sounds.
SAAS should use CS4 mode anyway, as otherwise sounds volume will be corrected twice.

**Comment by x87** [comment]
```
:SAAS_2530
4@ = Memory.GetVarPointer(18@)
memcpy 12@ {destination} 4@ {size} 48
13@ = AudioStream.GetVolume(22@)
Text.PrintFormattedNow("%.0f %.0f %.0f %s %.1f %i", 200, 10@, 11@, TIMERA, 12@, 13@, 31@)

```

no idea what triggers it yet

**Comment by x87** [comment]
list of changes between beta.1 and rc.1
https://github.com/cleolibrary/CLEO5/compare/v5.0.0-beta.1...v5.0.0-RC1

**Comment by x87** [comment]
I guess it's ini parsing https://github.com/cleolibrary/CLEO5/commit/3203afca76573ea6563e5246230bc6a2c39725a6

**Comment by x87** [comment]
```
:SAAS_917
1@ = 0
  3@ = IniFile.ReadInt("cleo\saas.ini", "SETTINGS", "turnOffRadio")
if 
  3@ == 1
goto_if_false @SAAS_993
set_local_var_bit_const 1@ {bitIndex} 2

:SAAS_993
  3@ = IniFile.ReadInt("cleo\saas.ini", "SETTINGS", "debugMode")
if 
  3@ == 1
goto_if_false @SAAS_1059
set_local_var_bit_const 1@ {bitIndex} 0
```

if `debugMode` is not present, value of 3@ is not changed and remains 1 (because turnOffRadio=1)

@MiranDMC

**Comment by MiranDMC** [comment]
> ```
> :SAAS_917
> 1@ = 0
>   3@ = IniFile.ReadInt("cleo\saas.ini", "SETTINGS", "turnOffRadio")
> if 
>   3@ == 1
> goto_if_false @SAAS_993
> set_local_var_bit_const 1@ {bitIndex} 2
> 
> :SAAS_993
>   3@ = IniFile.ReadInt("cleo\saas.ini", "SETTINGS", "debugMode")
> if 
>   3@ == 1
> goto_if_false @SAAS_1059
> set_local_var_bit_const 1@ {bitIndex} 0
> ```
> 
> if `debugMode` is not present, value of 3@ is [not changed](https://github.com/cleolibrary/CLEO5/commit/3203afca76573ea6563e5246230bc6a2c39725a6#diff-fd046931fd446f1c0499bd2bf69754a4d4e97aa90dcf6703a50bbb91a2554b90R69-R77) and remains 1 (because turnOffRadio=1)
> 
> @MiranDMC

Yes, as it is meant to be. Target value should be not changed, unless ReadInt succeed and returned condition result true.
In older CLEO  ReadFloat was working in that way, meanwhile ReadInt was returning 0x80000000 as indicator of error.
This behavior is preserved in legacy modes.

**Comment by x87** [comment]
@MiranDMC please write a doc that describes all places where the behavior was changed and CS4 mode can help. We throw in too much into "just rename to .cs4" and it's hard to track all cases.

Something like:

opcode xxxx
in CLEO4 returns X when not found
in CLEO5 keeps the same value, condition flag is false

opcode YYYY
in CLEO4 all arguments are required
in CLEO5 all arguments are optional

etc

This doc may help to triage the issues and for script authors to focus on specific problems

**Comment by x87** [comment]
@MiranDMC any progress on the above?

**Comment by MiranDMC** [comment]
> [@MiranDMC](https://github.com/MiranDMC) any progress on the above?

Not even started yet.

**Comment by MiranDMC** [comment]
> [@MiranDMC](https://github.com/MiranDMC) any progress on the above?

https://github.com/cleolibrary/CLEO5/wiki/CLEO4-Compat-Mode-(.cs4)
Still can be redacted, but seems to be complete list. Not so much after all.

**Comment by x87** [comment]
good start, we can go from there
I'll that link to the website
